projects/storefrontlib/src/layout/a11y/keyboard-focus/lock/lock-focus.directive.ts

```diff
@@ -86,63 +111,57 @@ export class LockFocusDirective extends TrapFocusDirective
     }
   }
 
-  ngAfterViewInit(): void {
-    if (!!this.group) {
+  ngAfterContentInit() {
+    if (this.shouldLock) {
       /**
        * If the component hosts a group of focusable children elmenents,
        * we persist the group key to the children, so that they can taken this
        * into account when they persist their focus state.
        */
-      this.service
-        .findFocusable(this.host)
-        .forEach(el =>
-          this.renderer.setAttribute(el, FOCUS_GROUP_ATTR, this.group)
-        );
-    }
-
-    /**
-     * When the component got (re)created, we lock the focusable children elements
-     * by settng the tabindex to `-1`.
-     */
-    if (this.isLocked) {
-      this.addTabindexToChildren(this.hasPersistedFocus ? 0 : -1);
-    }
+      if (!!this.group) {
+        this.service
+          .findFocusable(this.host)
+          .forEach(el =>
+            this.renderer.setAttribute(el, FOCUS_GROUP_ATTR, this.group)
+          );
+      }
 
-    if (this.isLocked && this.hasPersistedFocus) {
-      this.hasbeenUnlocked = true;
-      this.handleFocus(null);
-    } else {
-      super.ngAfterViewInit();
+      this.lockFocus();
     }
   }
 
-  /**
-   * Locks focus handling in case of a locked experience.
-   */
-  handleFocus(event: KeyboardEvent): void {
-    if (this.isLocked && !this.hasbeenUnlocked) {
-      this.addTabindexToChildren(-1);
-    } else {
-      super.handleFocus(event);
-      this.hasbeenUnlocked = false;
+  handleFocus(event?: KeyboardEvent): void {
+    if (this.shouldLock) {
+      this.lockFocus();
+
+      if (!event && this.service.hasPersistedFocus(this.host, this.config)) {
+        // Delay unlocking in case the host is using `ChangeDetectionStrategy.Default`
+        setTimeout(() => this.unlockFocus(event));
```

Comment:
Might I ask why at 137 you have a check for `!event`? I don't know if it's very useful.

Reply:
It's actually very important, but not clear. I'll refactor to make it apparent.